### PR TITLE
Unit tests for ActivityManager

### PR DESCRIPTION
**Pull request by barnettwilliam**
Some unit tests for ActivityManager.

A couple of minor changes were required for test-ability:

- Calls to global browser objects `document` and `window` should be via utility functions that can be stubbed.
- Imports have to be via objects and not direct export.  This is because ES6 protects exports which prevents them being spied on by Jasmine. It is a problem for all javascript test frameworks.
- ActivityManager constructor made costly http requests so these were separated out into initializeActivities() method that's called after it's created.

Code coverage for the module at approx 52%

### CODE REVIEW

File: platform/test/resources/TestUtility.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+export function configObjectEquals(configA, configB){
+
+    const strConfigA = JSON.stringify(configA);
+    const strConfigB = JSON.stringify(configB);
+
+    return (strConfigA === strConfigB)
+}
+
+/**
+ * Prints a configuration object including its values.
```
Comment (szschaler) [inline platform/test/resources/TestUtility.js:16]: The function doesn't strictly "print" the object, it just returns its stringification. Given that that's the case, is the function actually needed or could it just be replaced with direct uses of `JSON.stringify`?

File: platform/test/spec/testActivityManagerSpec.js [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+import { ActivityConfigValidator } from "../../src/ActivityConfigValidator.js";
+import { utility } from "../../src/Utility.js";
+import { ACTIVITY_2PANELS_1ACTION } from "../resources/TestActivityFiles.js";
+import { TOOL_1PANELDEF_1FUNCTION } from "../resources/TestToolFiles.js";
+import { configObjectEquals, configToString } from "../resources/TestUtility.js";
+
+
+describe("ActivityManager", () => {
+
+    it("can be created", () => {
```
Comment (szschaler) [inline platform/test/spec/testActivityManagerSpec.js:14]: Given that there are multiple tests for the constructor, does it make sense to group them into a nested `describe` and possibly even reuse the constructed object (with a `beforeAll`)?

**Review by szschaler — CHANGES_REQUESTED** [review]
Looks good. Just some minor suggestions.

File: platform/test/resources/TestUtility.js (hunk 1 above)
Comment (barnettwilliam) [reply platform/test/resources/TestUtility.js:16]: Yes it's not particularly useful as it is so will use stringify directly.

Comment (barnettwilliam) [reply platform/test/resources/TestUtility.js]: Removed in 52769f3d17c78207d5778a9ec46d8e090149bfb0

File: platform/test/spec/testActivityManagerSpec.js (hunk 2 above)
Comment (barnettwilliam) [reply platform/test/spec/testActivityManagerSpec.js]: Fixed in 52769f3d17c78207d5778a9ec46d8e090149bfb0

File: platform/test/spec/testActivityManagerSpec.js [hunk 3]
```diff
@@ -6,91 +6,67 @@ import { ActivityConfigValidator } from "../../src/ActivityConfigValidator.js";
...
-    it("constructor -  initialises the accessPanelDef property using param panelDefAccessor", () => {
-        // Setup
-        const fileh = new FileHandler("test://th.url") 
-        const refPanelDef = () => {};
+            spySearch = spyOn( utility, "getWindowLocationSearch");
 
-        // Call the target object
-        var am = new ActivityManager(refPanelDef , fileh);
+            // Call the target object
+            am = new ActivityManager(refPanelDef, fileh);
```
Comment (szschaler) [inline platform/test/spec/testActivityManagerSpec.js:30]: Should this be in each test (there is at least one test that has to call this again anyway...)?

File: platform/test/spec/testActivityManagerSpec.js [hunk 4]
```diff
@@ -6,91 +6,67 @@ import { ActivityConfigValidator } from "../../src/ActivityConfigValidator.js";
...
+        })
 
-    it ("constructor -  if the current activity is provided in the url query string, set the activityId property", () => {
-        // Setup
-        const ACTIVITY_URL="test://a.url";
-        const ACTIVITY_ID="a1"
-        const QUERY = "?"+ ACTIVITY_ID + "=&" + "activities=" + ACTIVITY_URL;
-        const fileh = new FileHandler("test://th.url") 
-        const refPanelDef = () => {};
+        it ("if the current activity is provided in the url query string, set the activityId property", () => {
```
Comment (szschaler) [inline platform/test/spec/testActivityManagerSpec.js:60]:
```suggestion
        it ("sets the activityId property if the current activity is provided in the url query string", () => {
```

File: platform/test/spec/testActivityManagerSpec.js [hunk 5]
```diff
@@ -6,91 +6,67 @@ import { ActivityConfigValidator } from "../../src/ActivityConfigValidator.js";
...
-        // Call the target object
-        var am = new ActivityManager(refPanelDef , fileh);
+            expect(am.activityId).toBe(ACTIVITY_ID);
+        })
 
-        // Check the expected results
-        expect(am.activityId).toBe(ACTIVITY_ID);
     })
 
     it("initializeActivities - causes the activity file to be fetched from its URL using the fileHandler", () => {
```
Comment (szschaler) [inline platform/test/spec/testActivityManagerSpec.js:72]: Shouldn't these be in the next `describe` as they are about activity initialisation?

**Review by szschaler — CHANGES_REQUESTED** [review]
Thanks, much improved. Some other questions/suggestions...

File: platform/test/spec/testActivityManagerSpec.js (hunk 3 above)
Comment (barnettwilliam) [reply platform/test/spec/testActivityManagerSpec.js]: Yes it does make each test uniform fixed in 023439982f88091a37a89567f047fcb5495458bd

File: platform/test/spec/testActivityManagerSpec.js (hunk 5 above)
Comment (barnettwilliam) [reply platform/test/spec/testActivityManagerSpec.js]: Yes the test style evolved slightly as they were added it makes sense to include them in the same describe block. Fixed in 023439982f88091a37a89567f047fcb5495458bd

File: platform/test/spec/testActivityManagerSpec.js (hunk 4 above)
Comment (barnettwilliam) [reply platform/test/spec/testActivityManagerSpec.js]: Updated in c1fc48ec710862c2619fcad8fc297215a00334e8

**Review by szschaler — APPROVED** [review]
LGTM

**Pull request merged**